run_project_tests.py

```diff
@@ -1045,61 +1074,103 @@ def __init__(self, category: str, subdir: str, skip: bool = False, stdout_mandat
     return gathered_tests
 
 def run_tests(all_tests: T.List[T.Tuple[str, T.List[TestDef], bool]],
-              log_name_base: str, failfast: bool,
-              extra_args: T.List[str], use_tmp: bool) -> T.Tuple[int, int, int]:
-    global logfile
+              log_name_base: str,
+              failfast: bool,
+              extra_args: T.List[str],
+              use_tmp: bool,
+              num_workers: int) -> T.Tuple[int, int, int]:
     txtname = log_name_base + '.txt'
     with open(txtname, 'w', encoding='utf-8', errors='ignore') as lf:
-        logfile = lf
-        return _run_tests(all_tests, log_name_base, failfast, extra_args, use_tmp)
+        return _run_tests(all_tests, log_name_base, failfast, extra_args, use_tmp, num_workers, lf)
+
+class TestStatus(Enum):
+    OK       = normal_green(' [SUCCESS] ')
+    SKIP     =       yellow(' [SKIPPED] ')
+    ERROR    =          red('  [ERROR]  ')
+    CANCELED =         cyan('[CANCELED] ')
+    RUNNING  =         blue(' [RUNNING] ')  # Should never be actually printed
+    LOG      =         bold('   [LOG]   ')  # Should never be actually printed
```

Comment:
this is incredibly hard to read because nothing is lined up except the = signs. There's litteraly *no* python style guid that recomends this, and this is *not* tablular data, can we just write this
```python
class TestStatus(Enum):
    OK = normal_green(' [SUCCESS] ')
    SKIP = yellow(' [SKIPPED] '`)
    ...
```

Reply:
For truely tabluar data this makes sense to me (things like the giant table of data in compilers/cuda.py), but really I don't care what the values of the enums are, the keys are much more intresting.